static/js/topic_generator.js

```diff
@@ -129,7 +130,7 @@ exports.next_topic = function (streams, get_topics, has_unread_messages, curr_st
     }
 
     var outer_gen = exports.fchain(stream_gen, get_topic_gen);
-    return outer_gen.next();
+    return { topic : outer_gen.next(), stream : selected_stream };
```

Comment:
I think it makes more sense here to have `get_topics` return a list of objects of the form `{stream_id: 1, topic: 'lunch'}`, so that you're not pulling out `selected_stream` as a side effect.  If you pass in full objects, you'll get out a full object.  And you can then also change the signature of `has_unread_messages` to take in an object of the form `{stream_id: 1, topic: 'lunch'}` as well.